integration/test/api_test.go

```diff
@@ -0,0 +1,242 @@
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000",
+			expectStatus: 404,
+		},
+		{
+			// not reliably testable
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000",
+			expectStatus: 500,
+			skip:         true,
+		},
+		{
+			// not reliably testable
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000",
+			expectStatus: 503,
+			skip:         true,
+		},
+		{
+			name:         "GET /v0/blob/{id}/status is 200",
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/" + testBlobResp.ID + "/status",
+			expectStatus: 200,
+		},
+		{
+			name:         "GET /v0/blob/{id}/status for unknown ID is 404",
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000/status",
+			expectStatus: 404,
+		},
+		{
+			// not reliably testable
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000/status",
+			expectStatus: 500,
+			skip:         true,
+		},
+		{
+			// not reliably testable
+			onMethod:     http.MethodGet,
+			onPath:       "/v0/blob/00000000-0000-0000-0000-000000000000/status",
+			expectStatus: 503,
+			skip:         true,
+		},
+	}
+
+	// Read and parse openapi.yaml for ensuring all paths, methods, and status
+	// codes are covered
+
+	schemaString, err := os.ReadFile("../../openapi.yaml")
+	require.NoError(t, err, "could not find openapi.yaml")
+
+	schemaMap := make(map[string]interface{})
+	err = yaml.Unmarshal(schemaString, schemaMap)
+	require.NoError(t, err)
+
+	var schemaCases []schemaCase
+
+	type kvmap = map[string]interface{}
+	for pathName, path := range schemaMap["paths"].(kvmap) {
+		for methodName, method := range path.(kvmap) {
+			for statusCode := range method.(kvmap)["responses"].(kvmap) {
+				schemaCases = append(schemaCases, schemaCase{
+					method:  methodName,
+					path:    pathName,
+					status:  statusCode,
+					covered: false,
+				})
+			}
+		}
+	}
+
+	// Run all tests
+	for _, test := range tests {
+		if !test.skip {
+			req, err := http.NewRequest(
+				test.onMethod,
+				requireJoinUrlPath(t, env.MotionAPIEndpoint, test.onPath),
+				bytes.NewReader([]byte(test.expectBody)),
+			)
+			req.Header.Set("Content-Type", test.onContentType)
+			require.NoError(t, err)
+
+			resp, err := http.DefaultClient.Do(req)
+			require.NoError(t, err)
+
+			// Body must be as expected
+			var body string
+			if test.expectBody != "" {
+				bodyBytes, err := io.ReadAll(resp.Body)
+				require.NoError(t, err)
+
+				body = string(bodyBytes)
+				require.Regexp(t, test.expectBody, string(body))
+			}
+
+			resp.Body.Close()
+
+			// Status code must be as expected
+			require.Equal(t, test.expectStatus, resp.StatusCode, "Incorrect status code for test %#v (resp body: %v)", test, body)
+		}
+
+		// Find matching schema case and mark as covered
+		for i := range schemaCases {
```

Comment:
I would do it differently
1. use client generation tool such as github.com/deepmap/oapi-codegen to generate the client code, this also validates the yaml file against openapi spec
2. Since we only have 3 APIs, it maybe overkill to check all API paths are covered so I would just rely on code coverage tool.

Reply:
Re 1. Do you think it's worth the increase in LOC? No one is asking for http client library for such simple api (yet) right?

Re 2. Either works. Considering the work is done already, and we want to focus on e2e tests first/more than vide coverage would it make sense to revisit this at later date and leave it as is for now?

Reply:
considering that everything works right now, and the code does and will continue to do its job for now, i'd say we can look at codegen client / coverage as a followup, as they are more heavy commitments compared to what i wrote so far

Reply:
sounds good